x/incentives/keeper/msg_server.go

```diff
@@ -64,3 +64,13 @@ func (server msgServer) AddToGauge(goCtx context.Context, msg *types.MsgAddToGau
 
 	return &types.MsgAddToGaugeResponse{}, nil
 }
+
+func (server msgServer) SetAutoStaking(goCtx context.Context, msg *types.MsgSetAutoStaking) (*types.MsgSetAutoStakingResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	err := server.keeper.SetAutostaking(ctx, &types.AutoStaking{
+		Address:              msg.Address,
```

Comment:
Great idea on just having an address just set their validator preference once. 

What I was thinking of was more complicated, this is definitely better!

Reply:
+1